test/e2e/performanceprofile/functests/2_performance_update_second/updating_profile_mcp_labels.go

```diff
@@ -0,0 +1,270 @@
+package __2_performance_update_second
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+	"strings"
+	"time"
+
+	. "github.com/onsi/ginkgo"
+	. "github.com/onsi/gomega"
+	machineconfigv1 "github.com/openshift/machine-config-operator/pkg/apis/machineconfiguration.openshift.io/v1"
+	corev1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/types"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+
+	performancev2 "github.com/openshift/cluster-node-tuning-operator/pkg/apis/performanceprofile/v2"
+	"github.com/openshift/cluster-node-tuning-operator/pkg/performanceprofile/controller/performanceprofile/components"
+	testutils "github.com/openshift/cluster-node-tuning-operator/test/e2e/performanceprofile/functests/utils"
+	testclient "github.com/openshift/cluster-node-tuning-operator/test/e2e/performanceprofile/functests/utils/client"
+	"github.com/openshift/cluster-node-tuning-operator/test/e2e/performanceprofile/functests/utils/discovery"
+	"github.com/openshift/cluster-node-tuning-operator/test/e2e/performanceprofile/functests/utils/mcps"
+	"github.com/openshift/cluster-node-tuning-operator/test/e2e/performanceprofile/functests/utils/nodes"
+	"github.com/openshift/cluster-node-tuning-operator/test/e2e/performanceprofile/functests/utils/profiles"
+)
+
+type checkFunction func(*corev1.Node) (string, error)
+
+var _ = Describe("[rfe_id:28761][performance] Updating parameters in performance profile", func() {
+	var profile, initialProfile *performancev2.PerformanceProfile
+	var performanceMCP string
+	var err error
+
+	chkCmdLine := []string{"cat", "/proc/cmdline"}
+	chkIrqbalance := []string{"cat", "/rootfs/etc/sysconfig/irqbalance"}
+
+	nodeLabel := testutils.NodeSelectorLabels
+	profile, err = profiles.GetByNodeLabels(nodeLabel)
+	Expect(err).ToNot(HaveOccurred())
+	initialProfile = profile.DeepCopy()
+
+	BeforeEach(func() {
+		if discovery.Enabled() && testutils.ProfileNotFound {
+			Skip("Discovery mode enabled, performance profile not found")
+		}
+
+		profile, err = profiles.GetByNodeLabels(nodeLabel)
+		Expect(err).ToNot(HaveOccurred())
+		performanceMCP, err = mcps.GetByProfile(profile)
+		Expect(err).ToNot(HaveOccurred())
+
+		// Verify that worker and performance MCP have updated state equals to true
+		for _, mcpName := range []string{testutils.RoleWorker, performanceMCP} {
+			mcps.WaitForCondition(mcpName, machineconfigv1.MachineConfigPoolUpdated, corev1.ConditionTrue)
+		}
+	})
+
+	Context("Updating of nodeSelector parameter and node labels", func() {
```

Comment:
This context takes around 40-50 min. Even if with this "patch" it works a refactor is required.